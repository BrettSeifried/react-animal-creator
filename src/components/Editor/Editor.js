import React from 'react';

import './Editor.css';

export default function Editor({ head, setHead, body, setBody, legs, setLegs, phrase, setPhrase }) {
  const handleClick = () => {};
  return (
    <div className="editor">
      <div className="form-control">
        <select value={head} name="head" onInput={(e) => setHead(e.target.value)}>
          <option value="">Select a Head</option>
          <option value="dog">Dog Head</option>
          <option value="bird">Bird Head</option>
          <option value="duck">Duck Head</option>
          <option value="horse">Horse Head</option>
        </select>
      </div>
      <div className="form-control">
        <select value={body} name="body" onInput={(e) => setBody(e.target.value)}>
          <option value="">Select a Body</option>
          <option value="blue">Blue Shirt</option>
          <option value="dress">Pretty Dress</option>
          <option value="pink">Pink Top</option>
          <option value="red">Red Top</option>
        </select>
      </div>
      <div className="form-control">
        <select value={legs} name="legs" onInput={(e) => setLegs(e.target.value)}>
          <option value="">Select Pants</option>
          <option value="blue">Blue Pants</option>
          <option value="dog">Dog legs</option>
          <option value="leg">No Pants</option>
          <option value="white">White Pants</option>
        </select>
      </div>
      <div className="form-control">
        <input
          type="text"
          value={phrase}
          onInput={(e) => setPhrase(e.target.value)}
          placeholder="Write a Catch Phrase"
        />
        <button onClick={(e) => setPhrase(e.target.value)}>Submit Phrase</button>
      </div>
    </div>
  );
}
