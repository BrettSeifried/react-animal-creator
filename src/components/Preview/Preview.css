main {
  color: whitesmoke;
  font-weight: bolder;
}
.character {
  display: flex;
  flex-direction: column;
  align-items: center;
}

img {
  height: 150px;
  width: auto;
}
