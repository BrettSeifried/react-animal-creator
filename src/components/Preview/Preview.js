import React from 'react';

import './Preview.css';

export default function Preview({ head, body, legs, phrase }) {
  return (
    <div className="character">
      <img alt={head} src={`${process.env.PUBLIC_URL}/assets/${head}-head.png`} />
      <img alt={body} src={`${process.env.PUBLIC_URL}/assets/${body}-middle.png`} />
      <img alt={legs} src={`${process.env.PUBLIC_URL}/assets/${legs}-pants.png`} />
    </div>
  );
}
