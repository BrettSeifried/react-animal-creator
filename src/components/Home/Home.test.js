import { render } from '@testing-library/react';
import Editor from '../Editor/Editor';
import Preview from '../Preview/Preview';

test('Ensuring Editor & Preview feed into Home.js', () => {
  const container = render(
    <>
      <Editor />
      <Preview />
    </>
  );
  expect(container).toMatchSnapshot();
});
