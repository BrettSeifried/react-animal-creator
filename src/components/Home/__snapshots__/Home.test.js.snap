// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ensuring Editor & Preview feed into Home.js 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="editor"
      >
        <div
          class="form-control"
        >
          <select
            name="head"
          >
            <option
              value=""
            >
              Select a Head
            </option>
            <option
              value="dog"
            >
              Dog Head
            </option>
            <option
              value="bird"
            >
              Bird Head
            </option>
            <option
              value="duck"
            >
              Duck Head
            </option>
            <option
              value="horse"
            >
              Horse Head
            </option>
          </select>
        </div>
        <div
          class="form-control"
        >
          <select
            name="body"
          >
            <option
              value=""
            >
              Select a Body
            </option>
            <option
              value="blue"
            >
              Blue Shirt
            </option>
            <option
              value="dress"
            >
              Pretty Dress
            </option>
            <option
              value="pink"
            >
              Pink Top
            </option>
            <option
              value="red"
            >
              Red Top
            </option>
          </select>
        </div>
        <div
          class="form-control"
        >
          <select
            name="legs"
          >
            <option
              value=""
            >
              Select Pants
            </option>
            <option
              value="blue"
            >
              Blue Pants
            </option>
            <option
              value="dog"
            >
              Dog legs
            </option>
            <option
              value="leg"
            >
              No Pants
            </option>
            <option
              value="white"
            >
              White Pants
            </option>
          </select>
        </div>
        <div
          class="form-control"
        >
          <input
            placeholder="Write a Catch Phrase"
            type="text"
            value=""
          />
          <button>
            Submit Phrase
          </button>
        </div>
      </div>
      <div
        class="character"
      >
        <img
          src="/assets/undefined-head.png"
        />
        <img
          src="/assets/undefined-middle.png"
        />
        <img
          src="/assets/undefined-pants.png"
        />
        <p>
           
           
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="editor"
    >
      <div
        class="form-control"
      >
        <select
          name="head"
        >
          <option
            value=""
          >
            Select a Head
          </option>
          <option
            value="dog"
          >
            Dog Head
          </option>
          <option
            value="bird"
          >
            Bird Head
          </option>
          <option
            value="duck"
          >
            Duck Head
          </option>
          <option
            value="horse"
          >
            Horse Head
          </option>
        </select>
      </div>
      <div
        class="form-control"
      >
        <select
          name="body"
        >
          <option
            value=""
          >
            Select a Body
          </option>
          <option
            value="blue"
          >
            Blue Shirt
          </option>
          <option
            value="dress"
          >
            Pretty Dress
          </option>
          <option
            value="pink"
          >
            Pink Top
          </option>
          <option
            value="red"
          >
            Red Top
          </option>
        </select>
      </div>
      <div
        class="form-control"
      >
        <select
          name="legs"
        >
          <option
            value=""
          >
            Select Pants
          </option>
          <option
            value="blue"
          >
            Blue Pants
          </option>
          <option
            value="dog"
          >
            Dog legs
          </option>
          <option
            value="leg"
          >
            No Pants
          </option>
          <option
            value="white"
          >
            White Pants
          </option>
        </select>
      </div>
      <div
        class="form-control"
      >
        <input
          placeholder="Write a Catch Phrase"
          type="text"
          value=""
        />
        <button>
          Submit Phrase
        </button>
      </div>
    </div>
    <div
      class="character"
    >
      <img
        src="/assets/undefined-head.png"
      />
      <img
        src="/assets/undefined-middle.png"
      />
      <img
        src="/assets/undefined-pants.png"
      />
      <p>
         
         
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
